### PR TITLE
[#2903]feat(spark-connector):  register specific catalog to Spark catalog manager

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
register specific catalog like `GravitinoHiveCatalog` to Spark catalog manager, no the general `GravitinoCatalog`

### Why are the changes needed?
The specific catalog could implement different interfaces like `FunctionCatalog`, to support Iceberg partition.
Fix: #2903 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
existing tests

### CODE REVIEW

**Comment by FANNG1** [comment]
@jerryshao @qqqttt123 @caican00 please help to review when you are free, thanks

File: spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java [hunk 1]
```diff
@@ -58,16 +59,48 @@ public void shutdown() {
...
+      default:
+        LOG.warn("Skip registering {} because {} is not supported yet.", catalogName, provider);
+        return;
+    }
+
+    String sparkCatalogConfigName = "spark.sql.catalog." + catalogName;
+    Preconditions.checkArgument(
+        !sparkConf.contains(sparkCatalogConfigName),
+        catalogName + " is already registered to SparkCatalogManager");
+    sparkConf.set(sparkCatalogConfigName, catalogClassName);
```
Comment (caican00) [inline spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java:102]: remove `Preconditions.checkArgument` and call `sparkConf.setIfMissing`?

**Comment by caican00** [comment]
maybe `GravitinoHiveCatalog` and `GravitinoIcebergCatalog` are better placed on the corresponding code path, such as `com.datastrato.gravitino.spark.connector.catalog.hive.GravitinoHiveCatalog` and `com.datastrato.gravitino.spark.connector.catalog.iceberg.GravitinoIcebergCatalog`

File: spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java [hunk 2]
```diff
@@ -58,16 +59,48 @@ public void shutdown() {
...
+      case "lakehouse-iceberg":
+        catalogClassName = GravitinoIcebergCatalog.class.getName();
+        break;
+      default:
+        LOG.warn("Skip registering {} because {} is not supported yet.", catalogName, provider);
+        return;
+    }
+
+    String sparkCatalogConfigName = "spark.sql.catalog." + catalogName;
+    Preconditions.checkArgument(
```
Comment (caican00) [inline spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java:99]: why add this limitation?

**Comment by FANNG1** [comment]
> maybe `GravitinoHiveCatalog` and `GravitinoIcebergCatalog` are better placed on the corresponding code path, such as `com.datastrato.gravitino.spark.connector.catalog.hive.GravitinoHiveCatalog` and `com.datastrato.gravitino.spark.connector.catalog.iceberg.GravitinoIcebergCatalog`

ok

File: spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java (hunk 1 above)
Comment (caican00) [reply spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/plugin/GravitinoDriverPlugin.java:102]: ok.

**Comment by FANNG1** [comment]
@qqqttt123 do you have time to review this?

**Comment by FANNG1** [comment]
@jerryshao do you have time to review this?

**Comment by caican00** [comment]
Hi @jerryshao @qqqttt123 could you help review this PR? we need to implement `FunctionCatalog`  based on this PR to support iceberg partitions. Thank you very much.

**Review by jerryshao — APPROVED** [review]
LGTM.

**Pull request merged**